autogpt_platform/autogpt_libs/autogpt_libs/utils/cache.py

```diff
@@ -255,22 +396,47 @@ def sync_wrapper(*args: P.args, **kwargs: P.kwargs):
 
         # Add cache management methods
         def cache_clear() -> None:
-            cache_storage.clear()
+            if shared_cache:
+                # Clear only cache keys (prefixed with "cache:") using pipeline for efficiency
+                keys = list(redis_client.scan_iter("cache:*", count=100))
+                if keys:
+                    pipeline = redis_client.pipeline()
+                    for key in keys:
+                        pipeline.delete(key)
+                    pipeline.execute()
+            else:
+                cache_storage.clear()
 
         def cache_info() -> dict[str, int | None]:
-            return {
-                "size": len(cache_storage),
-                "maxsize": maxsize,
-                "ttl_seconds": ttl_seconds,
-            }
+            if shared_cache:
+                # Count only cache keys
+                cache_keys = list(redis_client.scan_iter("cache:*"))
```

Comment:
* is actually a good idea to remove the in memory cache too, instead of us removing page by page and randomly remove 20 we can expose an api to clear the api using *? obviously not mandatory here just a suggestion.